x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/EsPhysicalOperationProviders.java

```diff
@@ -165,7 +168,11 @@ public final PhysicalOperation sourcePhysicalOperation(EsQueryExec esQueryExec,
         assert esQueryExec.estimatedRowSize() != null : "estimated row size not initialized";
         int rowEstimatedSize = esQueryExec.estimatedRowSize();
         int limit = esQueryExec.limit() != null ? (Integer) esQueryExec.limit().fold() : NO_LIMIT;
-        if (sorts != null && sorts.isEmpty() == false) {
+        boolean metadataScoreSet = esQueryExec.attrs()
+            .stream()
+            .anyMatch(a -> a instanceof MetadataAttribute && a.name().equals(SCORE_FIELD.getName()));
+        boolean scoring = EsqlCapabilities.Cap.METADATA_SCORE.isEnabled() && metadataScoreSet;
```

Comment:
This is problematic and release builds (like serverless) would fail the execution with an exception.

> What would happen if the user requests the _score field, but METADATA_SCORE was not enabled?

I went ahead and tried it: `FROM books METADATA _score` fails with:
```
java.lang.IllegalArgumentException: Cannot project block with index [9] from a page with size [9]
	at org.elasticsearch.compute.data.Page.projectBlocks(Page.java:280)
	at org.elasticsearch.compute.operator.ProjectOperator.process(ProjectOperator.java:51)
	at org.elasticsearch.compute.operator.AbstractPageMappingOperator.getOutput(AbstractPageMappingOperator.java:76)
	at org.elasticsearch.compute.operator.Driver.runSingleLoopIteration(Driver.java:258)
	at org.elasticsearch.compute.operator.Driver.run(Driver.java:189)
	at org.elasticsearch.compute.operator.Driver$1.doRun(Driver.java:378)
	at org.elasticsearch.server@9.0.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)
	at org.elasticsearch.server@9.0.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:34)
	at org.elasticsearch.server@9.0.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1023)
	at org.elasticsearch.server@9.0.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)\n

```

Reply:
I think this should now be fixed, as part of [a0a1150](https://github.com/elastic/elasticsearch/pull/113120/commits/a0a1150753a82060831cb835ba51ef8e8e025305). So this would fail same as any other non-existing metadata field, if not in snapshot builds.